homeassistant/components/samsungtv/bridge.py

```diff
@@ -42,7 +42,7 @@ def get_bridge(method, host, port=None, token=None):
 
     def __init__(self, method, host, port):
         """Initialize Bridge."""
-        self.port = port
+        self.port = 8001 if port is None else port
```

Comment:
Shouldn't we be using the port that was selected in `try_connect` and saved in the config entry? How can port be `None` at the stage reported in the issue?

Reply:
@MartinHjelmare the config entry has been created prior my last PR. Old config flow didn't store the port on the entry. Since my only TVs running on port 8001 were supported. That's the reason behind my fix (avoid the user to recreate the config entry)

Reply:
Instead of handling that as you did now, you can also check on init if the port is missing in the configuration entry and update it.

For example <https://github.com/home-assistant/core/blob/dev/homeassistant/components/wled/__init__.py#L54>.

Reply:
Yeah, I think it's better to make sure that port exists in the config entry.

If it doesn't exist we should probably check with the bridge which port is correct before saving the port.